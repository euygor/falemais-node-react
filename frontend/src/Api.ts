import axios from "axios";

const BASE = "http://localhost:3333";

const axiosInstance = axios.create({
    baseURL: BASE
});

export const api = {
    getFees: async () => {
        const response = await axiosInstance.get(`/values`);
        return response;
    }
}