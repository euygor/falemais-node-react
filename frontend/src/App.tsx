import styles from './App.module.css';
import { Form } from './components/Form';
import imageTelephone from './assets/images/telephone.png';


const App = () => {
  return (
    <div className={styles.container}>
      <div className={styles.leftSide}>
        <img src={imageTelephone} alt="" />
        <p><strong>Os melhores planos com as melhores tarifas vocÃª encontra AQUI !</strong></p>
      </div>
      <div className={styles.rightSide}>
        <Form />
      </div>
    </div>
  );
}

export default App;