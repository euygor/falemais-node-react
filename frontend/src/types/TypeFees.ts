export type TypeFees = {
    origin: number;
    destination: number;
    price: number;
}