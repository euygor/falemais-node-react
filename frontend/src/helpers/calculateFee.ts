import { TypeFees } from './../types/TypeFees';

export const calculateFee = (origin: number, destination: number, plan: number, minutes: number, fees: TypeFees[]) => {

    let fee = fees.find(fee => fee.origin === origin && fee.destination === destination);

    let surplus = minutes % plan;

    let result = fee && minutes > plan ? ((minutes * fee.price) - (plan * fee.price)) + (plan !== 0 ? ((10 * fee.price) / 100) * surplus : 0) : 0;

    return result;
} 