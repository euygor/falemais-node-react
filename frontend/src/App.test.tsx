import App from './App';
import { render, screen } from '@testing-library/react';
import { calculateFee } from './helpers/calculateFee';
import { TypeFees } from './types/TypeFees';
import { api } from './Api';


// Integration test
describe('Integration test', () => {
  it('check if component "<Form />" exists within the application looking for the name "FaleMais"', () => {
    render(<App />);
    const text = screen.getByText('FaleMais');
    expect(text).toBeInTheDocument();
  });

  it('check if api call is working', async () => {
    const response = await api.getFees();
    expect(response.status).toBe(200);
  });

});

// Unit test
describe('Unit test', () => {
  it('should render correctly', () => {
    const { asFragment } = render(<App />);
    expect(asFragment()).toMatchSnapshot();
  });

  it('should calculate the fee correctly', async () => {
    const fees: TypeFees[] = await (await api.getFees()).data;
    const inputVoid = calculateFee(0, 0, 0, 0, fees);
    const inputFilled = Number(calculateFee(11, 18, 30, 32, fees).toFixed(2));

    expect(inputVoid).toBe(0);
    expect(inputFilled).toBe(3.89);
  });
});