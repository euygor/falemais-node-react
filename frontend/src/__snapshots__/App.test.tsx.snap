// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Unit test should render correctly 1`] = `
<DocumentFragment>
  <div
    class="container"
  >
    <div
      class="leftSide"
    >
      <img
        alt=""
        src="telephone.png"
      />
      <p>
        <strong>
          Os melhores planos com as melhores tarifas você encontra AQUI !
        </strong>
      </p>
    </div>
    <div
      class="rightSide"
    >
      <div
        class="form"
      >
        <fieldset>
          <legend>
            <h4>
              FaleMais
            </h4>
          </legend>
          <select>
            <option
              value="0"
            >
              Selecione a Origem
            </option>
            <option
              value="11"
            >
              DDD - ( 11 )
            </option>
            <option
              value="16"
            >
              DDD - ( 16 )
            </option>
            <option
              value="17"
            >
              DDD - ( 17 )
            </option>
            <option
              value="18"
            >
              DDD - ( 18 )
            </option>
          </select>
          <select>
            <option
              value="0"
            >
              Selecione o Destino
            </option>
            <option
              value="11"
            >
              DDD - ( 11 )
            </option>
            <option
              value="16"
            >
              DDD - ( 16 )
            </option>
            <option
              value="17"
            >
              DDD - ( 17 )
            </option>
            <option
              value="18"
            >
              DDD - ( 18 )
            </option>
          </select>
          <select>
            <option
              value="0"
            >
              Selecione um Plano
            </option>
            <option
              value="30"
            >
              Plano FaleMais 30
            </option>
            <option
              value="60"
            >
              Plano FaleMais 60
            </option>
            <option
              value="120"
            >
              Plano FaleMais 120
            </option>
          </select>
          <input
            placeholder="Informe os minutos"
            type="number"
            value=""
          />
        </fieldset>
        <div
          class="result"
        >
          <p>
            Origem: ( 0 )
          </p>
          <p>
            Destino: ( 0 )
          </p>
          <p>
            Sem Plano: R$ 0,00
          </p>
          <p>
            Com Plano: R$ 0,00
          </p>
        </div>
        <div
          class="info"
        >
          <p>
            OBS: Ligações com DDD de Origem e Destino iguais não são cobradas as tarifas.
          </p>
        </div>
      </div>
    </div>
  </div>
</DocumentFragment>
`;
